src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,237 @@
+        public bool Equals(System.Buffers.NRange other) { throw null; }
+        public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? value) { throw null; }
+        public override int GetHashCode() { throw null; }
+        public (nint Offset, nint Length) GetOffsetAndLength(nint length) { throw null; }
+        public static explicit operator checked System.Range (System.Buffers.NRange value) { throw null; }
+        public static explicit operator System.Range (System.Buffers.NRange value) { throw null; }
+        public static implicit operator System.Buffers.NRange (System.Range range) { throw null; }
+        public static System.Buffers.NRange StartAt(System.Buffers.NIndex start) { throw null; }
+        public System.Range ToRange() { throw null; }
+        public System.Range ToRangeUnchecked() { throw null; }
+        public override string ToString() { throw null; }
+    }
+}
 namespace System.Numerics.Tensors
 {
+    public partial interface ITensor<TSelf, T> : System.Collections.Generic.IEnumerable<T>, System.Collections.IEnumerable, System.IEquatable<TSelf>, System.Numerics.IEqualityOperators<TSelf, TSelf, bool> where TSelf : System.Numerics.Tensors.ITensor<TSelf, T>
+    {
+        static TSelf? Empty { get { throw null; } }
+        bool IsEmpty { get; }
+        bool IsPinned { get; }
+        ref T this[scoped System.ReadOnlySpan<nint> indices] { get; }
+        int Rank { get; }
+        System.ReadOnlySpan<nint> Strides { get; }
+        System.Numerics.Tensors.ReadOnlyTensorSpan<T> AsReadOnlyTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        void Clear();
+        void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+        void Fill(T value);
+        ref T GetPinnableReference();
+        static abstract bool operator ==(TSelf left, TSelf right);
+        static abstract bool operator ==(TSelf left, T right);
+        static abstract implicit operator System.Numerics.Tensors.ReadOnlyTensorSpan<T> (TSelf value);
+        static abstract implicit operator System.Numerics.Tensors.TensorSpan<T> (TSelf value);
+        static abstract bool operator !=(TSelf left, TSelf right);
+        static abstract bool operator !=(TSelf left, T right);
+        TSelf Slice(scoped System.ReadOnlySpan<System.Buffers.NRange> ranges);
+        bool TryCopyTo(System.Numerics.Tensors.TensorSpan<T> destination);
+    }
+    public readonly ref partial struct ReadOnlyTensorSpan<T>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        [System.CLSCompliantAttribute(false)]
+        public unsafe ReadOnlyTensorSpan(void* pointer, System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides = default(System.ReadOnlySpan<nint>)) { throw null; }
```

Comment:
There's notably several constructors missing here and this overload in particular needs to be `T* data, nint dataLength, ...`

We then don't want to default strides and want separate overloads instead